pymmmc/clv/models/beta_geo.py

```diff
@@ -0,0 +1,324 @@
+        with pm.Model() as self.model:
+            a = self.model.register_rv(a_prior, name="a")
+            b = self.model.register_rv(b_prior, name="b")
+
+            alpha = self.model.register_rv(alpha_prior, name="alpha")
+            r = self.model.register_rv(r_prior, name="r")
+
+            def logp(t_x, x, a, b, r, alpha, T):
+                """
+                The log-likelihood expression here aligns with expression (4) from [3]
+                due to the possible numerical instability of expression (3).
+                """
+                zero_observations = at.eq(x, 0)
+
+                A = betaln(a, b + x) - betaln(a, b) + at.gammaln(r + x) - at.gammaln(r)
+                A += r * at.log(alpha) - (r + x) * at.log(alpha + T)
+
+                B = (
+                    betaln(a + 1, b + x - 1)
+                    - betaln(a, b)
+                    + at.gammaln(r + x)
+                    - at.gammaln(r)
+                )
+                B += r * at.log(alpha) - (r + x) * at.log(alpha + t_x)
+
+                logp = at.switch(
+                    zero_observations,
+                    A,
+                    at.logaddexp(A, B),
+                )
+
+                return check_parameters(
+                    logp,
+                    a > 0,
+                    b > 0,
+                    alpha > 0,
+                    r > 0,
+                    msg="a, b, alpha, r > 0",
+                )
+
+            pm.Potential(
+                "likelihood",
+                logp(x=frequency, t_x=recency, a=a, b=b, alpha=alpha, r=r, T=T),
+            )
+
+    def _process_priors(self, a_prior, b_prior, alpha_prior, r_prior):
+        # hyper priors for the Gamma params
+        if a_prior is None:
+            a_prior = pm.HalfFlat.dist()
+        else:
+            assert a_prior.eval().shape == ()
+        if b_prior is None:
+            b_prior = pm.HalfFlat.dist()
+        else:
+            assert b_prior.eval().shape == ()
+
+        # hyper priors for the Beta params
+        if alpha_prior is None:
+            alpha_prior = pm.HalfFlat.dist()
+        else:
+            assert alpha_prior.eval().shape == ()
+        if r_prior is None:
+            r_prior = pm.HalfFlat.dist()
+        else:
+            assert r_prior.eval().shape == ()
+
+        a_prior.str_repr = types.MethodType(str_for_dist, a_prior)
+        b_prior.str_repr = types.MethodType(str_for_dist, b_prior)
+        alpha_prior.str_repr = types.MethodType(str_for_dist, alpha_prior)
+        r_prior.str_repr = types.MethodType(str_for_dist, r_prior)
+
+        return a_prior, b_prior, alpha_prior, r_prior
+
+    def _to_xarray(self, *arrays, **kwargs):
+        if "coord_label" in kwargs:
+            coord_label = kwargs["coord_label"]
+        else:
+            coord_label = "customer_id"
+
+        num_customers = len(arrays[0])
+        dims = (coord_label,)
+        coords = {coord_label: range(num_customers)}
```

Comment:
We should ask users to pass the `customer_id`. If they want the expected_num_of_purchases of customers 10 to 100, those should be the coords, not 0 - 90

Reply:
The sampled parameters aren't specific to an individual if I remember correctly. What effect will these coords have on the output other than renaming the DataArray coords?

Reply:
Do you mean that we want to have `expected_num_purchases` to have a method signature that takes input only `customer_id`? Are you saying that we want to provide users to option to call

```python
expected_num_purchases(some_t_of_interest, customer_id=range(10, 90))
```
instead of
```python
expected_num_purchases(some_t_of_interest, frequency[10:90], recency[10:90], T[10:90])
```

Reply:
Or are you suggesting that we ask users to provide `customer_id` to clarify the coords of the DataArray? Sort of like what is being done in the Gamma-Gamma model currently

Reply:
> Or are you suggesting that we ask users to provide `customer_id` to clarify the coords of the DataArray? Sort of like what is being done in the Gamma-Gamma model currently

This